tests/test_flatten.py

```diff
@@ -1,33 +1,274 @@
-from pytest import raises
+from copy import deepcopy
```

Comment:
Why `deepcopy` instead of [`dict.copy`](https://docs.python.org/3/library/stdtypes.html#dict.copy)?

Reply:
Because copy, as it is mentioned int he documentation:
> Return a shallow copy of the dictionary.
So, even using it (copy()) the test fail because of the use of `DICT_DATA_WITH_NESTED_VALUES_IN_NESTED_COLUMNS` in two differents tests that alter its structure.
I understood, it copies the data strucutre.
I understand that `copy()` creates a new object but keeps references with the original. And because of that, changes made in the copy affects thte original.

The [`deepcopy`](https://python.readthedocs.io/en/v2.7.2/library/copy.html#copy.deepcopy), however, creates a full copy of the data:
> A deep copy constructs a new compound object and then, recursively, inserts copies into it of the objects found in the original.

Reply:
Nice!